rfcs/55-entropy.md

```diff
@@ -0,0 +1,135 @@
+
+The default source of world entropy [`Entropy::default()`] is non-deterministic and seeded from the operating system. It is guarenteed to be suitable for [cryptographic applications](https://en.wikipedia.org/wiki/Pseudorandom_number_generator#Cryptographic_PRNGs) but the actual seeding mechanism is an implementation detail that will change over time.
+
+You may choose to determinstically seed your own world entropy via [`Entropy::from`]. The  seed you choose may have security implications or influence the distribution of the resulting random numbers. See https://rust-random.github.io/book/guide-seeding.html for more details about how to pick a "good" random seed for your needs.
+
+Depending on your game and the type of randomness you require, when specifying a seed you would normally do one of the following:
+
+1. Get a good random seed out-of-band and hardcode it in the source.
+2. Dynamically call to the OS and print the seed so the user can rerun deterministically. In games like [Factorio](https://www.factorio.com/) sharing random seeds is encouraged and supported.
+3. Dynamically call to the OS and share the seed with a server so the client and server deterministically execute together.
+4. Load the seed from a server so the client and server deterministically execute together.
+
+
+## Implementation strategy
+
+https://github.com/bevyengine/bevy/pull/2504
+
+- The PR includes `rand` as a dependency but it is not in public API. We might be able to slim it down to [`getrandom`](https://crates.io/crates/getrandom).
+
+
+## Drawbacks
+
+- This may not be general enough to include in Bevy.
+- This may not be general enough to be on by default.
+- Includes `rand` as a dependency
+  - But not in public API.
+
+## Rationale and alternatives
+
+- Why is this design the best in the space of possible designs?
+  - It gives flexibility for PRNG crates while also enforcing standards on the ecosystem. 
+  - It defaults to what someone would resonably expect if they wrote it themselves.
+  - It defaults to something safe (suitable for cryptographic functions), removing a possible footgun.
+- What other designs have been considered and what is the rationale for not choosing them?
+  - We could go higher and expose an API closer to `rand`. This is what [Godot](https://docs.godotengine.org/en/stable/tutorials/math/random_number_generation.html), [Unity](https://docs.unity3d.com/ScriptReference/Random.html), and [Unreal](https://docs.unrealengine.com/4.27/en-US/BlueprintAPI/Math/Random/) do. We are not experts in PRNG API design, and while `rand` is clearly the most popular random crate in the Rust ecosystem we don't currently want to tie bevy to any particluar API in case something better emerges.
```

Comment:
```suggestion
  - We could go higher and expose an API closer to `rand`. This is what [Godot](https://docs.godotengine.org/en/stable/tutorials/math/random_number_generation.html), [Unity](https://docs.unity3d.com/ScriptReference/Random.html), and [Unreal](https://docs.unrealengine.com/4.27/en-US/BlueprintAPI/Math/Random/) do. We are not experts in PRNG API design, and while `rand` is clearly the most popular random crate in the Rust ecosystem we don't currently want to tie bevy to any particular API in case something better emerges.
```